django/template/base.py

```diff
@@ -216,11 +217,12 @@ def render(self, context):
 
 
 class Token(object):
-    def __init__(self, token_type, contents):
+    def __init__(self, token_type, contents, position=None, lineno=None):
         # token_type must be TOKEN_TEXT, TOKEN_VAR, TOKEN_BLOCK or
         # TOKEN_COMMENT.
         self.token_type, self.contents = token_type, contents
-        self.lineno = None
+        self.lineno = lineno
+        self.position = position
```

Comment:
A comment clarifying the type of `position` would help. I assume that:

- in debug mode, it's a `(start, end)` tuple
- in normal mode, it's ` None`